app/src/main/java/com/sma2/sma2/SignalRecording/CSVFileWriter.java

```diff
@@ -0,0 +1,97 @@
+package com.sma2.sma2.SignalRecording;
+
+import android.os.Environment;
+import android.util.Log;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Locale;
+
+public class CSVFileWriter {
+    private String TAG = "CSVFileWriter";
+    private final String DELIMITER = ";";
+    private final String NEW_LINE = "\r\n";
+    private final String FILE_ENDING =  ".csv";
+
+    private BufferedWriter mBufferedWriter = null;
+
+    private final String PATH = Environment.getExternalStorageDirectory() + "/Apkinson/";
+
+    private File openFile(String path, String fileName) {
+        // check if directory exists and if not create it
+        File directory = new File(path);
+        if (! directory.exists()) {
+            Log.d(TAG, "Directory did not exists, creating: " + path);
+            directory.mkdir();
+        }
+
+        // Try to create File
+        File file = new File(path + fileName);
+        if (!file.exists()) {
+            try {
+                Log.d(TAG,"Try to create new File at: " + path + fileName);
+                file.createNewFile();
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+            Log.d(TAG,"File Successfully created at: " + path + fileName);
+        } else {
+            Log.d(TAG, "File: " + path + fileName + " already exists!");
+        }
+        return file;
+    }
+
+    public CSVFileWriter() {
+        String path = PATH;
+        String currTime = new SimpleDateFormat("yyyy-MM-dd_HH-mm", Locale.getDefault()).format(new Date());
+
+        File file = openFile(path, currTime + FILE_ENDING);
+
+        try {
+            mBufferedWriter = new BufferedWriter(new FileWriter(file));
+        } catch (Exception e) {
+            Log.e(TAG,"ERROR opening file");
```

Comment:
I don't think we should handle this exception here. And we shouldn't log the stack trace...